contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+        address acct;
+        // v component of permit signature
+        uint8 v;
+        // r component of permit signature
+        bytes32 r;
+        // s component of permit signature
+        bytes32 s;
+    }
+
+    // struct defining the upper and lower boundaries of notional for an asset
+    struct MinMaxNotional {
+        // minimum notional value allowed
+        uint256 min;
+        // maximum notional value allowed
+        uint256 max;
+    }
+
+    ///@dev mapping between an asset address to a struct consisting of uint256 min and a uint256 max which will be its notional floor and cap size allowed
+    mapping(address => MinMaxNotional) public minMaxNotional;
+
+    ///@notice mapping between order id and order details
+    mapping(uint256 => Order) public orders;
+
+    ///@notice mapping between order id and order status
+    mapping(uint256 => OrderStatus) public orderStatus;
+
+    ///@notice mapping between a Market Maker address and its whitelist status
+    mapping(address => bool) public isWhitelistedMarketMaker;
+
+    ///@notice mapping between an asset address and its corresponding fee
+    mapping(address => uint256) public fee;
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice constructor related to ERC2771
+     * @param _trustedForwarder trusted forwarder address
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(MinimalForwarder _trustedForwarder) ERC2771ContextUpgradeable(address(_trustedForwarder)) {}
+
+    /**
+     * @notice initialize the deployed contract
+     * @param _beneficiary beneficiary address
+     * @param _addressBook AddressBook address
+     * @param _owner owner address
+     * @param _fillDeadline fill deadline duration
+     * @param _usdc USDC address
+     */
+    function initialize(
+        address _addressBook,
+        address _owner,
+        address _beneficiary,
+        uint256 _fillDeadline,
+        address _usdc
+    ) external initializer {
+        require(_addressBook != address(0), "OTCWrapper: addressbook address cannot be 0");
+        require(_owner != address(0), "OTCWrapper: owner address cannot be 0");
+        require(_beneficiary != address(0), "OTCWrapper: beneficiary address cannot be 0");
+        require(_fillDeadline > 0, "OTCWrapper: fill deadline cannot be 0");
+        require(_usdc != address(0), "OTCWrapper: usdc address cannot be 0");
+
+        __Ownable_init();
+        __ReentrancyGuard_init();
+
+        addressbook = AddressBookWrapperInterface(_addressBook);
+        marginRequirements = MarginRequirementsWrapperInterface(addressbook.getMarginRequirements());
+        controller = ControllerWrapperInterface(addressbook.getController());
+        oracle = OracleWrapperInterface(addressbook.getOracle());
+        whitelist = WhitelistWrapperInterface(addressbook.getWhitelist());
+
```

Comment:
we should either add getMarginPool() and getOtokenFactory() and remove addressbook 

OR

just use addressbook on the fly for everything (like getMarginRequirements(), getController(), getOracle(), getWhitelist()) and remove marginRequirements, controller, oracle, whitelist state variables.

I am leaning towards latter.